src/core_modules/capture-core/components/WidgetEnrollmentEventNew/AccessVerification/AccessVerification.component.js

```diff
@@ -1,10 +1,13 @@
 // @flow
 import React from 'react';
+import { useRulesEngineOrgUnit } from '../../../hooks/useRulesEngineOrgUnit';
 import { Validated } from '../Validated';
 import { NoAccess } from './NoAccess.component';
 import type { Props } from './accessVerification.types';
 
-export const AccessVerificationComponent = ({ eventAccess, onCancel, widgetReducerName, ...passOnProps }: Props) => {
+export const AccessVerificationComponent = ({ eventAccess, onCancel, orgUnitId, ...passOnProps }: Props) => {
+    const orgUnit = useRulesEngineOrgUnit(orgUnitId);
```

Comment:
But the code would become quite a lot more loosely coupled, and straight forward, if I could just add a component right above `Validated.container` with only one mission, namely creating the orgUnit object that `Validated` needs. Two reasons against it:

1. It will block subsequent components from rendering.
2. It adds bloat to the component tree.

The first point is of little consequence as global reasoning reveals that I've already made `useLifecycle` set `ready` to false if `orgUnit` is undefined, which in turn blocks most (but not all) rendering of presumably visible components.
The second point is less important than the clarity gained using this approach, I think (simplicity of code outweighs simplicity of component tree?).
I will do it the way I want first, then you can decide if you prefer it done differently after seeing the result.

Reply:
Your points are valid, I would still prefer `Validated.container`, but it is negligible, so I'm fine with leaving it in its own component. In general, I've found that a happy medium between a bloated component tree and more coupled components is the way to go (especially after the introduction of hooks). Makes it more readable.